index.js

```diff
@@ -289,6 +289,12 @@ Compiler.prototype = {
    */
 
   visitNode: function(node){
+    if (!node) {
+      throw new TypeError('node is ' + node);
```

Comment:
I think this could be a little more descriptive, e. g., `'invalid node received: ' + node`